pdns/dnsdistdist/tcpiohandler.cc

```diff
@@ -595,7 +621,7 @@ class GnuTLSTicketsKey
 #ifdef HAVE_LIBSODIUM
       sodium_munlock(d_key.data, d_key.size);
```

Comment:
maybe `safe_memzero` should always be called and in that function `HAVE_LIBSODIUM` is checked?

Reply:
That doesn't feel right because `sodium_munlock()` does more than just zeroing the buffer.